index.js

```diff
@@ -618,13 +645,29 @@ const isAdminOrUser = (user) => {
 		if(category !== "all") {
 			whereValues.push(category)
 			whereConditions.push(`main.category_to_cases.category_id = $${whereValues.length}`);
+			whereConditionsRevisedQ.push(`main.category_to_cases.category_id = $${whereValues.length}`);
 		}
 
+
+
 		if(req.query.caseSetId) {
 			const case_set = await client.query(`SELECT * FROM ugc.random_case_sets WHERE id = $1`, [req.query.caseSetId])
 			whereConditions.push(`m.id IN ('${case_set.rows[0].case_set.map(c => c.id).join('\',\'')}')`);
+			whereConditionsRevisedQ.push(`c.id IN ('${case_set.rows[0].case_set.map(c => c.id).join('\',\'')}')`);
 		}
 
+		let revisedQ = `
```

Comment:
No need to use `revisedQ` and comment things - please replace the existing variables where necessary.